diff -burN net-snmp-5.7.1/agent/mibgroup/host/data_access/swinst_pkginfo.c net-snmp-5.7.1-swinst_pkginfo/agent/mibgroup/host/data_access/swinst_pkginfo.c
--- net-snmp-5.7.1/agent/mibgroup/host/data_access/swinst_pkginfo.c	2011-09-27 23:53:47.000000000 -0500
+++ net-snmp-5.7.1-swinst_pkginfo/agent/mibgroup/host/data_access/swinst_pkginfo.c	2012-03-14 18:03:54.000000000 -0500
@@ -140,6 +140,7 @@
         memcpy( entry->swDate, cp, date_len );
         entry->swDate_len = date_len;
     }
+    if (d != NULL)
     closedir( d );
 
     DEBUGMSGTL(("swinst:load:arch"," loaded %d entries\n",
