--- net-snmp-5.7.1/agent/helpers/table.c	2011-09-27 23:53:47.000000000 -0500
+++ net-snmp-5.7.1-saved/agent/helpers/table.c	2012-05-26 14:08:44.000000000 -0500
@@ -785,6 +785,7 @@
                  */
                 DEBUGMSGT(("sparse", "retry for NOSUCHOBJECT\n"));
                 table_info = netsnmp_extract_table_info(request);
+                if (table_info != NULL) {
                 table_info->colnum = netsnmp_table_next_column(table_info);
                 if (0 != table_info->colnum) {
                     memcpy(coloid, reginfo->rootoid,
@@ -795,8 +796,7 @@
                                        coloid, reginfo->rootoid_len + 2);
                     
                     request->requestvb->type = ASN_PRIV_RETRY;
-                }
-                else {
+                    } else {
                     /*
                      * If we don't have column info, reset to null so
                      * the agent will move on to the next table.
@@ -806,6 +806,7 @@
             }
         }
     }
+    }
     return status;
 }
 
