--- a/agent/mibgroup/if-mib/ifXTable/ifXTable.c 2012-02-15 16:12:53.000000000 +0200
+++ b/agent/mibgroup/if-mib/ifXTable/ifXTable.c 2012-02-15 16:17:20.000000000 +0200
@@ -1390,13 +1390,6 @@
 
     netsnmp_assert(NULL != rowreq_ctx);
 
-    if (0 == rowreq_ctx->data.ifConnectorPresent)
-        return MFD_SKIP;
-
-    /*
-     * TODO:231:o: |-> Extract the current value of the ifConnectorPresent data.
-     * copy (* ifConnectorPresent_val_ptr ) from rowreq_ctx->data
-     */
     (*ifConnectorPresent_val_ptr) = rowreq_ctx->data.ifConnectorPresent ?
         TV_TRUE : TV_FALSE;
 

