diff -burN net-snmp-5.7.1/agent/mibgroup/host/data_access/swrun_procfs_status.c net-snmp-5.7.1-swrun_procfs_status/agent/mibgroup/host/data_access/swrun_procfs_status.c
--- net-snmp-5.7.1/agent/mibgroup/host/data_access/swrun_procfs_status.c	2011-09-27 23:53:47.000000000 -0500
+++ net-snmp-5.7.1-swrun_procfs_status/agent/mibgroup/host/data_access/swrun_procfs_status.c	2012-03-14 17:52:47.000000000 -0500
@@ -94,10 +94,14 @@
         }
         fclose(fp);
 
-        for ( cp = buf; *cp != ':'; cp++ )
+        for ( cp = buf; *cp && *cp != ':'; cp++ )
             ;
-        while (isspace(*(++cp)))	/* Skip ':' and following spaces */
+        while (*cp && isspace(*(++cp)))	/* Skip ':' and following spaces */
             ;
+        if (!*cp) {
+            netsnmp_swrun_entry_free(entry);
+            continue;
+        }
         entry->hrSWRunName_len = snprintf(entry->hrSWRunName,
                                    sizeof(entry->hrSWRunName)-1, "%s", cp);
         if ( '\n' == entry->hrSWRunName[ entry->hrSWRunName_len-1 ]) {
@@ -172,9 +176,9 @@
         fclose(fp);
 
         cp = buf;
-        while ( ' ' != *(cp++))    /* Skip first field */
+        while ( *cp && ' ' != *(cp++))    /* Skip first field */
             ;
-        while ( ' ' != *(cp++))    /* Skip second field */
+        while ( *cp && ' ' != *(cp++))    /* Skip second field */
             ;
         
         switch (*cp) {
@@ -189,21 +193,18 @@
         default:   entry->hrSWRunStatus = HRSWRUNSTATUS_INVALID;
                    break;
         }
-        for (i=10; i; i--) {   /* Skip STATUS + 10 fields */
-            while (' ' != *(cp++))
+        for (i=11; i; i--) {   /* Skip STATUS + 10 fields */
+            while (*cp && ' ' != *(cp++))
                 ;
-            cp++;
         }
         entry->hrSWRunPerfCPU  = atoi( cp );   /*  utime */
-        while ( ' ' != *(cp++))
+        while (*cp &&  ' ' != *(cp++))               /* Skip utime */
             ;
-        cp++;				   /* Skip utime */
         entry->hrSWRunPerfCPU += atoi( cp );   /* +stime */
 
-        for (i=8; i; i--) {   /* Skip stime + 8 fields */
-            while (' ' != *(cp++))
+        for (i=9; i; i--) {   /* Skip stime + 8 fields */
+            while (*cp && ' ' != *(cp++))
                 ;
-            cp++;
         }
         entry->hrSWRunPerfMem  = atoi( cp );   /*  rss */
         entry->hrSWRunPerfMem *= (getpagesize()/1024);  /* in kB */
