--- net-snmp-5.7-clean/snmplib/snmp_api.c	2011-07-01 17:35:46.000000000 -0500
+++ net-snmp-5.7/snmplib/snmp_api.c	2011-07-07 09:24:00.686337553 -0500
@@ -834,7 +834,6 @@
     register_mib_handlers();
     register_default_handlers();
     init_snmp_transport();
-    init_snmpv3(type);
     init_snmp_alarm();
     init_snmp_enum(type);
     init_vacm();
@@ -899,7 +898,6 @@
 void
 snmp_shutdown(const char *type)
 {
-    snmp_store(type);
     snmp_call_callbacks(SNMP_CALLBACK_LIBRARY, SNMP_CALLBACK_SHUTDOWN, NULL);
     shutdown_snmp_logging();
     snmp_alarm_unregister_all();
